### PR TITLE
Spec for `ErrorInvalidOpcode` state

### PR DESCRIPTION
**Pull request by silathdiir**
Upstream issue https://github.com/privacy-scaling-explorations/zkevm-specs/issues/350

Referenced `go-ethereum` code, this error is returned in [opUndefined method](https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go#L813), and it is set to [JumpTable code](https://github.com/ethereum/go-ethereum/blob/master/core/vm/jump_table.go#L1040).

So this error could be returned at [code 1](https://github.com/ethereum/go-ethereum/blob/master/core/vm/interpreter.go#L184) and [code 2](https://github.com/ethereum/go-ethereum/blob/master/core/vm/interpreter.go#L234) in `evm.interpreter.Run` method.

And `evm.interpreter.Run` method could be invoked via [CALL](https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L227), [CALLCODE](https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L284), [DELEGATECALL](https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L324), [STATICCALL](https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L380)  [CREATE](https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L450) (and `CREATE2`).

### CODE REVIEW

File: src/zkevm_specs/evm/execution/error_invalid_opcode.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from ..table import CallContextFieldTag
+
+
+def invalid_opcode(instruction: Instruction):
+    # Fixed lookup for invalid opcode.
+    opcode = instruction.opcode_lookup(True)
+    instruction.responsible_opcode_lookup(opcode)
+
+    # Constrain it is an invalid opcode.
+    instruction.constrain_in(opcode, [FQ(i) for i in invalid_opcodes()])
```
Comment (silathdiir) [inline src/zkevm_specs/evm/execution/error_invalid_opcode.py:14]: I could not confirm if need to constrain it is an invalid opcode here.

File: src/zkevm_specs/evm/execution/error_invalid_opcode.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from ...util import FQ
+from ..execution_state import ExecutionState
+from ..instruction import Instruction, Transition
+from ..opcode import invalid_opcodes
+from ..table import CallContextFieldTag
+
+
+def invalid_opcode(instruction: Instruction):
+    # Fixed lookup for invalid opcode.
+    opcode = instruction.opcode_lookup(True)
```
Comment (DreamWuGit) [inline src/zkevm_specs/evm/execution/error_invalid_opcode.py:10]: in this case, opcode should not be true code,      `opcode = instruction.opcode_lookup(False)`  should be correct.

File: src/zkevm_specs/evm/execution/error_invalid_opcode.py (hunk 1 above)
Comment (DreamWuGit) [reply src/zkevm_specs/evm/execution/error_invalid_opcode.py:14]: in the soec we can check like this, think about how to check in real rust circuit.

File: src/zkevm_specs/evm/execution/error_invalid_opcode.py (hunk 2 above)
Comment (silathdiir) [reply src/zkevm_specs/evm/execution/error_invalid_opcode.py]: Fixed to `instruction.opcode_lookup(False)`.

File: specs/error_state/ErrorInvalidOpcode.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+1. Current opcode must be invalid.
+2. Current call must be failed.
+3. If it's a root call, it transits to `EndTx`, and the call's `IsPersistent` must be false
+4. if it is not root call, it restores caller's context by reading to `rw_table`, then does step state transition to it.
+
+### Lookups
+
+- 1 Call Context lookup `CallContextFieldTag.IsSuccess`.
+- 1 Call Context lookup `CallContextFieldTag.IsPersistent` (Only if Current instruction is root.).
```
[the remark this answers is not in the document]
Comment (silathdiir) [reply specs/error_state/ErrorInvalidOpcode.md]: Sorry. Deleted `IsPersistent` related in Markdown.

File: src/zkevm_specs/evm/execution/error_invalid_opcode.py (hunk 1 above)
Comment (silathdiir) [reply src/zkevm_specs/evm/execution/error_invalid_opcode.py:14]: Need to update this code according to circuit.

**Comment by silathdiir** [comment]
Thanks for dream's code review, moved to upstream https://github.com/privacy-scaling-explorations/zkevm-specs/pull/362

**Pull request closed without merging**